File: backend/src/api/submissions/submissions.controller.ts
import { Body, Controller, Get, Post, UseGuards } from '@nestjs/common'
import { SubmissionsService } from '../../application/submissions/submissions.service'
import { SubmissionInputDto, type SubmissionDto } from '../../contracts/dto/submission.dto'
import { CurrentUser, JwtAuthGuard, type AuthUser } from '../auth/jwt-auth.guard'

@Controller('submissions')
@UseGuards(JwtAuthGuard)
export class SubmissionsController {
  constructor(private readonly submissions: SubmissionsService) {}

  @Post()
  create(@CurrentUser() user: AuthUser, @Body() input: SubmissionInputDto): Promise<SubmissionDto> {
    return this.submissions.create(input, user.id)
  }

  @Get('mine')
  mine(@CurrentUser() user: AuthUser): Promise<SubmissionDto[]> {
    return this.submissions.listMine(user.id)
  }
}

File: backend/src/main/java/ee/gemspot/api/web/SubmissionsController.java
package ee.gemspot.api.web;

import ee.gemspot.api.common.CurrentUser;
import ee.gemspot.api.dto.SubmissionDto;
import ee.gemspot.api.dto.SubmissionInputDto;
import ee.gemspot.api.service.SubmissionsService;
import jakarta.validation.Valid;
import java.util.List;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;

/** Auth-gated. POST creates a PENDING submission (201, Nest default). */
@RestController
@RequestMapping("/submissions")
public class SubmissionsController {

    private final SubmissionsService submissions;

    public SubmissionsController(SubmissionsService submissions) {
        this.submissions = submissions;
    }

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public SubmissionDto create(@Valid @RequestBody SubmissionInputDto input) {
        return submissions.create(input, CurrentUser.id());
    }

    @GetMapping("/mine")
    public List<SubmissionDto> mine() {
        return submissions.listMine(CurrentUser.id());
    }
}
